### ISSUE TITLE
Actuator metrics: Path vars not recognized e.g. if spring security check fails - number of metrics explode

### ISSUE THREAD

**Issue by Jan-M** [spring-projects/spring-boot#4377]
If you have spring security enabled and e.g. use Oauth2 and query an endpoint with a path variable, e.g. `/api/account/{accountid}/data` and the Oauth2 validation fails the actuator metrics will track a new metric as the path variable is only handled later in the filter chain ("BEST_MATCHING_PATTERN_ATTRIBUTE" is not available yet).

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/MetricsFilter.java#L151

```
# Path /api/account/{accountid}/data

/api/account/1/data 
#yields gauge.response.api.account.1.data and  counter.status.200.api.account.1.data

/api/account/2/data
...

/api/account/3/data
...
```

With this you can a make the /metric endpoint return lots of data and you can with invalid credentials consume more and more memory.

Similar to how 404s are handled the unknown case should fall back to something like "unmapped", avoiding the increasing number of metrics.

**Comment by wilkinsona** [comment]
@Jan-M Can you describe the steps you're taking to produce a 200 response even though the OAuth2 validation has failed? Using `spring-boot-sample-secure-oauth2`, I've tried accessing `http://localhost:8080/flights/1` with no access token and with an invalid token. In both cases I receive a 401 response and the `counter.status.401.unmapped` metric is incremented.

**Comment by Jan-M** [comment]
Sorry, seems I was not precise enough when writing this, looking at our internal issue from back then it is obviously not a 200 status, but a 500, and the metrics entry is as follows.

```
"counter.status.500.api.abcdef.1234567890.ghijklm": 1
```

From my colleague who debugged this for our own filter, that uses and exposes on dropwizard metrics timer per URL I did get the following feedback:

The "ZmonMetricsFilter" is not installed for the error chain
The "ZmonMetricsFilter" runs before the OAuth2AuthenticationProcessingFilter
BEST_MATCHING_PATTERN_ATTRIBUTE is set in the ServletHandler (which runs after the filter chain)

Our "ZmonMetricsFilter" is a blunt reimplementation and as I saw the above entries also for the spring boot metrics so I wanted to share this issue with you. I can at the moment not shed any light on how the oauth2 filter/check was implemented, maybe it was a custom filter. If I get you right this is maybe not an issue if spring security implementation is used.
